plugins/nf-wave/src/main/io/seqera/wave/plugin/WaveClient.groovy

```diff
@@ -338,6 +357,25 @@ class WaveClient {
             }
         }
 
+        /*
+         * If 'spack' directive is specified use it to create a Dockefile
+         * to assemble the target container
+         */
+        Path spackFile = null
+        if( attrs.spack ) {
+            if( dockerScript )
+                throw new IllegalArgumentException("Unexpected spack and dockerfile conflict")
```

Comment:
Good point! However, I added this one by blindly copying the `conda` case above, i.e. `L347-L348` match `L366-L367`. I haven't tested the following,but I actually think these checks should be taken off, because `attrs` has already been validated above at `L327-L330`:
```
        if( config().strategy() )
            attrs = resolveConflicts(attrs, config().strategy())
        else
            checkConflicts(attrs, task.lazyName())
```

I would let you approve whether to delete or not.

Reply:
melius abundare .. :D ok, let's just add `Unexpected spack and dockerfile conflict while resolving wave container`, same for conda

Reply:
agree! done!